File: frontend/src/types/dept.ts
// 部门类型定义
export interface Dept {
  id?: number
  name: string
  createTime?: string
  updateTime?: string
}

// 分页结果类型
export interface PageResult<T> {
  total: number
  rows: T[]
}

// API 响应类型
export interface ApiResponse<T = any> {
  code: number
  message: string
  data: T
}

File: springboot-web-02/src/main/java/com/itheima/pojo/Dept.java
package com.itheima.pojo;

import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Dept {
    private Integer id;

    // 部门名称长度限制与数据库 varchar(10) 保持一致，超出时返回带限制值的错误提示
    @Size(max = 10, message = "部门名称长度不能超过 {max} 个字符")
    private String name;

    private LocalDateTime createTime;
    private LocalDateTime updateTime;
}
